src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -558,10 +569,39 @@ private void getCanvasResponse(HttpServletRequest request,
 			sb.append(line);
 		}
 
+		if( url.substring(url.indexOf("/api")).matches(appExtPropertiesFile.getProperty(CANVAS_API_GET_COURSE_ENROLL)) && request.getSession().getAttribute(LAUNCH_TYPE).equals("lti")){
+			addEnrollmentsToSession(request, sb);
+		}
+
 		out.print(sb.toString());
 		out.flush();
 	}
 
+	private void addEnrollmentsToSession(HttpServletRequest request,
+			StringBuilder sb) {
+		M_log.debug("ENROLLMENTS: " + sb.toString());
+		HashMap<Integer, String> enrollmentsFound =  new HashMap<Integer, String>();
+		JSONArray enrollmentsArray = new JSONArray(sb.toString());
+		for(int i = 0; i < enrollmentsArray.length(); i++){
+			JSONObject childJSONObject = enrollmentsArray.getJSONObject(i);
+			M_log.debug("ENROLLMENT RECORD: " + childJSONObject.get("course_id") + " " + childJSONObject.get("course_section_id") + " " + childJSONObject.get("type"));
+			if(enrollmentsFound.containsKey(childJSONObject.getInt("course_id"))){
+				String oldType = enrollmentsFound.get(childJSONObject.getInt("course_id"));
+				String newType = childJSONObject.getString("type");
+				M_log.debug("OLD TYPE: " + oldType);
+				M_log.debug("NEW TYPE: " + newType);
+				if(enrollmentsMap.get(newType) > enrollmentsMap.get(oldType)){
+					enrollmentsFound.put(childJSONObject.getInt("course_id"), childJSONObject.getString("type"));
```

Comment:
Just a thought. If two conditions do the identical action and all other states just skip that action would it be clearer to expect the action will happen and test whether or not to skip it?